ucsmsdk/ucssession.py

```diff
@@ -118,6 +121,40 @@ def evt_channel(self):
     def last_update_time(self):
         return self.__last_update_time
 
+    def _freeze(self):
+        save = {
+            "ip": self.__ip,
+            "username": self.__username,
+            "password": self.__password,
+            "proxy": self.__proxy,
+            "uri": self.__uri,
+            "ucs": self.__ucs,
+            "name": self.__name,
+            "cookie": self.__cookie,
+            "session_id": self.__session_id,
+            "version": self.__version,
+            "refresh_period": self.__refresh_period,
+            "priv": self.__priv,
+            "domains": self.__domains,
+            "channel": self.__channel,
+            "evt_channel": self.__evt_channel,
+            "last_update_time": self.__last_update_time,
+            "force": self.__force,
+            "auto_refresh": self.__auto_refresh,
+            "dump_xml": self.__dump_xml,
+            "redirect": self.__redirect
+        }
+        return json.dumps(save)
+
+    def _unfreeze(self, params_json):
+        params_dict = json.loads(params_json)
+        for param in params_dict:
+            setattr(self, '_UcsSession__' + param, params_dict[param])
+
+        # cookie might be stale, if so relogin
+        if self.__auto_refresh:
```

Comment:
so, if you look at the `_refresh` method. it tries a `login` when `refresh` fails. In the scenario that you have mentioned, the refresh would fail and a login would be tried and that would fail too.